assets/aws/single-ami.pkr.hcl

```diff
@@ -0,0 +1,206 @@
+
+  # sanitise the AMI name so that its safe for use with AWS 
+  ami_name = regex_replace(local.unsafe_ami_name, "[^a-zA-Z0-9\\- \\(\\).\\'[\\]@]", "-")
+
+  # split the comma-separated region list out into a proper array
+  destination_regions = [for s in split(",", var.ami_destination_regions) : trimspace(s)]
+
+  ami_description = "Teleport${var.teleport_fips ? " with FIPS support" : ""} using Hardened Amazon Linux 2023 AMI"
+  build_type      = "production${var.teleport_fips ? "-fips" : ""}"
+
+  # Used in AWS access policies. Do not change without consulting the teleport-prod 
+  # terraform.
+  resource_purpose_tag_value = "release-ami-builder"
+
+  # We can't execute from /tmp due to CIS hardenning guidelines, so we have to 
+  # specify a place to run provisioning scripts
+  remote_folder = "/home/ec2-user"
+}
+
+source "amazon-ebs" "teleport-aws-linux" {
+  ami_description                           = local.ami_description
+  ami_name                                  = local.ami_name
+  ami_regions                               = local.destination_regions
+  associate_public_ip_address               = true
+  temporary_security_group_source_public_ip = true
+  force_delete_snapshot                     = true
+  instance_type                             = var.aws_instance_type
+  region                                    = var.aws_region
+  encrypted_boot = false
+  run_tags = {
+    Name    = local.ami_name
+    purpose = local.resource_purpose_tag_value
+  }
+  run_volume_tags = {
+    Name = local.ami_name
+  }
+  snapshot_tags = {
+    Name = local.ami_name
+  }
+  source_ami   = data.amazon-ami.teleport-hardened-base.id
+  ssh_pty      = true
+  ssh_username = "ec2-user"
+  subnet_filter {
+    filters = {
+      "tag:purpose"     = local.resource_purpose_tag_value
+      "tag:environment" = "prod"
+    }
+    most_free = true
+  }
+  tags = {
+    BuildTimestamp      = var.ami_build_timestamp
+    BuildType           = "production"
+    Name                = local.ami_name
+    TeleportVersion     = var.teleport_version
+    TeleportEdition     = var.teleport_type
+    TeleportFipsEnabled = var.teleport_fips
+  }
+  vpc_id = var.aws_vpc_id
+}
+
+build {
+  sources = [
+    "source.amazon-ebs.teleport-aws-linux",
+  ]
+
+  provisioner "shell" {
+    remote_folder = local.remote_folder
+    inline = ["mkdir /tmp/files"]
+  }
+
+  provisioner "file" {
+    source      = "files/"
+    destination = "/tmp/files"
+  }
+
+  provisioner "shell" {
+    remote_folder = local.remote_folder
+    inline = [
+      "sudo cp /tmp/files/system/* /etc/systemd/system/",
+      "sudo cp /tmp/files/bin/* /usr/local/bin/"
+    ]
+  }
+
+  provisioner "shell" {
+    remote_folder = local.remote_folder
+    environment_vars = [
+      "FIPS=${var.teleport_fips ? 1 : 0}"
+    ]
+    inline = [
+      "if [ \"$FIPS\" -eq 1 ]; then touch /tmp/teleport-fips; fi"
```

Comment:
I don't understand why you need to use a file "flag" (`/tmp/teleport-fips`) and not just an environment variable. Could you just set the `FIPS` env var in the next provisioner block and check for that var in the script?

Reply:
Inertia, really. I felt like I'd already made so many changes that I didn't want make any more changes - but you're right. I'll check that none of the `systemd` units expect it, and then update.